source/common/network/dns_impl.cc

```diff
@@ -1,97 +1,126 @@
 #include "dns_impl.h"
 
 #include "common/common/assert.h"
-#include "common/event/libevent.h"
 #include "common/network/address_impl.h"
 #include "common/network/utility.h"
 
+#include "ares.h"
 #include "event2/event.h"
 
 namespace Network {
 
-DnsResolverImpl::DnsResolverImpl(Event::DispatcherImpl& dispatcher) : dispatcher_(dispatcher) {
-  // This sets us up to receive signals on an fd when async DNS resolved are completed.
-  sigset_t mask;
-  sigemptyset(&mask);
-  sigaddset(&mask, Event::Libevent::Global::DNS_SIGNAL_ID);
-  signal_fd_ = signalfd(-1, &mask, SFD_NONBLOCK);
-  RELEASE_ASSERT(-1 != signal_fd_);
-
-  event_assign(&signal_read_event_, &dispatcher_.base(), signal_fd_,
-               EV_READ | EV_PERSIST, [](evutil_socket_t, short, void* arg) -> void {
-                 static_cast<DnsResolverImpl*>(arg)->onSignal();
-               }, this);
-
-  event_add(&signal_read_event_, nullptr);
+DnsResolverImpl::DnsResolverImpl(Event::DispatcherImpl& dispatcher)
```

Comment:
Can we make this take Event::Dispatcher vs. Event::DispatcherImpl. Should no longer be dependent on libevent. Also please remove _impl includes for the dispatcher.